### ISSUE TITLE
Unit testing application using flask-classful

### ISSUE THREAD

**Issue by dbalgley** [teracyhq/flask-classful#116]
Hello all,

I am working on a project that uses flask-classful for our API routing and I'm currently attempting to write unit tests for said APIs. I seem to be running into a problem, however.

We use the Resource Representation implementation detailed in the doc (http://flask-classful.teracy.org/#adding-resource-representations-get-real-classy-and-put-on-a-top-hat) for all our Views, but when I attempt to use the pytest/flask method of testing my API functions, I get an error from werkzeug/wrappers.py:59:

```
TypeError: 'dict' object is not callable
The view function did not return a valid response. The return type must be a string, tuple, Response instance, or WSGI callable, but it was a dict.
```

My best guess is that for whatever reason I am bypassing the representation section of our code and I'm therefore not returning a werkzeug-compliant Response-type object (which is generated by calling flask.make_response(...). I'm certain something I'm doing is wrong I'm just not sure what part-test setup, api, or view. I have tried putting the representation code as what my API returns and that seems to return something that makes my tests pass (and not get that error) but it's pretty radically different from what all the guides seem to talk about as far as how to write the API function with flask-classful and the test with pytest. Additionally, calling jsonify() or Response() on my response seems to pass the tests as well, but once again we should be doing that in the output_json function right?

Is there some guide/document I can read that would break down how to unit test APIs with flask-classful in the mix? Could anyone take a look at this and tell me what I've done wrong? Any help would be appreciated as I'm at wit's-end here.

My View (simplified):
```python
# The location of the Resource Representation code from the docs
from our_stuff.api import representations as rep
class FooView(flask_classful.FlaskView):
    representations = {'application/json': rep.output_json}
    ...
    # noinspection PyMethodMayBeStatic
    @flask_jwt_extended.jwt_required
    def index(self):
        # Grab the info from the db
        data = ...
        # jsonify our data using marshmallow schemas
        response = schemas.schema_jsonify('foo', data)
        # return our response with 200
        return response, 200
````

My test (simplified) is as follows:
```python
class TestFoo(object):
    test_foo_index(self, client):
        # Code that sets up the auth headers we need for our flask_jwt_required decorator
        headers = ...
        # Access the API
        response = client.get('/api/foo/', headers=headers)
        assert response.status_code is 200
````

My client fixture in conftest.py:
```python
def client(app):
    # Read this was necessary for flask-classful testing
    app.testing = True
    with app.test_client() as client:
        yield client
````
(For reference, I set up the app in another fixture which sets up the flask app, including registering the views)

Any help would be appreciated, and if you need more information I'm happy to provide. I know there are a lot of variables and things that could be going wrong here but really just a guide as to how to get representations to represent correctly when testing, or a guide as to how to test while using flask-classful would be helpful. Thank you!

**Comment by hoatle** [comment]
@dbalgley I guess you can refer to this unit test for your tests https://github.com/teracyhq/flask-classful/blob/develop/test_classful/test_representations.py hopefully it helps.

**Comment by dbalgley** [comment]
@hoatle you're a hero-It turns out the problem was in my header that I was passing...I thought it wasn't important when I showed my code here but the Content-Type and Accept fields need to be in the headers you pass to the client.get() call if you're expecting to use a Resource Representation like json. I looked through the unit tests you guys had up and didn't notice that the first time and copying the header data you had made it work! We're up and running. Thank you so much!

````python
headers = {
            'Authorization': 'Bearer {}'.format(access_token),
            'Content-Type': 'application/json',
            'Accept': 'application/json',
        }
````
